server/src/main/java/org/elasticsearch/index/translog/TranslogHeader.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.translog;
+
+import org.apache.lucene.codecs.CodecUtil;
+import org.apache.lucene.index.CorruptIndexException;
+import org.apache.lucene.index.IndexFormatTooNewException;
+import org.apache.lucene.index.IndexFormatTooOldException;
+import org.apache.lucene.store.InputStreamDataInput;
+import org.apache.lucene.store.OutputStreamDataOutput;
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.io.Channels;
+import org.elasticsearch.common.io.stream.InputStreamStreamInput;
+import org.elasticsearch.common.io.stream.OutputStreamStreamOutput;
+
+import java.io.IOException;
+import java.nio.channels.FileChannel;
+import java.nio.file.Path;
+
+/**
+ * Each translog file is started with a translog header then followed by translog operations.
+ */
+final class TranslogHeader {
+    public static final String TRANSLOG_CODEC = "translog";
+
+    public static final int VERSION_CHECKSUMS    = 1; // pre-2.0 - unsupported
+    public static final int VERSION_CHECKPOINTS  = 2; // added checkpoints
+    public static final int VERSION_PRIMARY_TERM = 3; // added primary term
+    public static final int CURRENT_VERSION = VERSION_PRIMARY_TERM;
+
+    public static final long UNKNOWN_PRIMARY_TERM = -1L;
+
+    private final String translogUUID;
+    private final long primaryTerm;
+    private final int headerSizeInBytes;
+
+    TranslogHeader(String translogUUID, long primaryTerm) {
```

Comment:
can we add java docs about what the uuid and term mean? i.e., what do they enforce?